### PR TITLE
Fix #1770 : Marquee restarts after submitting Answer

### LINKED ISSUE #1770: Marquee restarts after submitAnswer
**Describe the bug**

**To Reproduce**
1. Go to `The Meaning of Equal Parts` exploration
2. Go to `Second state`. Note that the title is not moving.
3. Submit current answer.
4. Marquee effect starts.

**Expected behavior**
Marquess effect should start only on click on the toolbar and not automatically.

### PR DESCRIPTION
**Pull request by prayutsu**
## Explanation

Fixes #1770 
Sample test added for a review

## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

**Comment by anandwana001** [comment]
Could you add more on this, why this fails, what approaches you had taken and what places you had doubts? 
It will be helpful for me to comment on this.

**Comment by prayutsu** [comment]
@anandwana001 
Screenshot from 2020-09-16 01-15-58
Screenshot from 2020-09-16 01-16-09

**Comment by prayutsu** [comment]
> Could you add more on this, why this fails, what approaches you had taken and what places you had doubts?
> It will be helpful for me to comment on this.

Actually I am trying to check that after clicking submit button, is ```exploration_toolBar_title.isSeleceted``` is true or not

**Comment by prayutsu** [comment]
@anandwana001 I also wrote another test in ```ExplorationActivityTest```. Ans this is the error - 
```
androidx.test.espresso.PerformException: Error performing 'single click' on view 'with id: org.oppia.app:id/continue_button'.
at androidx.test.espresso.PerformException$Builder.build(PerformException.java:86)
at androidx.test.espresso.base.DefaultFailureHandler.getUserFriendlyError(DefaultFailureHandler.java:87)
at androidx.test.espresso.base.DefaultFailureHandler.handle(DefaultFailureHandler.java:59)
at androidx.test.espresso.ViewInteraction.waitForAndHandleInteractionResults(ViewInteraction.java:322)
at androidx.test.espresso.ViewInteraction.desugaredPerform(ViewInteraction.java:178)
at androidx.test.espresso.ViewInteraction.perform(ViewInteraction.java:119)
at org.oppia.app.player.exploration.ExplorationActivityTest.testMarquee_openRatioExploration_continueToInteraction_submitAnswer_checkToolBarTitleIsSelected(ExplorationActivityTest.kt:645)
at java.lang.reflect.Method.invoke(Native Method)
at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
at androidx.test.internal.runner.junit4.statement.RunBefores.evaluate(RunBefores.java:80)
at androidx.test.internal.runner.junit4.statement.RunAfters.evaluate(RunAfters.java:61)
at androidx.test.rule.ActivityTestRule$ActivityStatement.evaluate(ActivityTestRule.java:527)
at org.junit.rules.RunRules.evaluate(RunRules.java:20)
at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
at androidx.test.ext.junit.runners.AndroidJUnit4.run(AndroidJUnit4.java:104)
at org.junit.runners.Suite.runChild(Suite.java:128)
at org.junit.runners.Suite.runChild(Suite.java:27)
at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
at org.junit.runner.JUnitCore.run(JUnitCore.java:115)
at androidx.test.internal.runner.TestExecutor.execute(TestExecutor.java:56)
at androidx.test.runner.AndroidJUnitRunner.onStart(AndroidJUnitRunner.java:392)
at android.app.Instrumentation$InstrumentationThread.run(Instrumentation.java:2145)
Caused by: java.lang.RuntimeException: Action will not be performed because the target view does not match one or more of the following constraints:
at least 90 percent of the view's area is displayed to the user.
Target view: "AppCompatButton{id=2131296468, res-name=continue_button, visibility=VISIBLE, width=277, height=132, has-focus=false, has-focusable=true, has-window-focus=true, is-clickable=true, is-enabled=true, is-focused=false, is-focusable=true, is-layout-requested=false, is-selected=false, layout-params=androidx.constraintlayout.widget.ConstraintLayout$LayoutParams@7497d71, tag=null, root-is-layout-requested=false, has-input-connection=false, x=627.0, y=44.0, text=Continue, input-type=0, ime-target=false, has-links=false}"
at androidx.test.espresso.ViewInteraction.doPerform(ViewInteraction.java:252)
at androidx.test.espresso.ViewInteraction.access$100(ViewInteraction.java:65)
at androidx.test.espresso.ViewInteraction$1.call(ViewInteraction.java:158)
at androidx.test.espresso.ViewInteraction$1.call(ViewInteraction.java:155)
at java.util.concurrent.FutureTask.run(FutureTask.java:266)
at android.os.Handler.handleCallback(Handler.java:873)
at android.os.Handler.dispatchMessage(Handler.java:99)
at android.os.Looper.loop(Looper.java:193)
at android.app.ActivityThread.main(ActivityThread.java:6669)
at java.lang.reflect.Method.invoke(Native Method)
at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)

```

**Comment by anandwana001** [comment]
As per the @rt4914 comment, You should begin with the first point. Writing test cases in the first place will not be a good idea. Also, you had only written the test cases, I didn't find the solution/fix for the issue. 
As per the tests, you had added, I am adding a review on those, which can help you afterwards when you write test cases. 

```
1. First try to find the exact reasons of the bugs.
2. Once you have identified the reason, write a test case which actually fails because of the bug. (This will be needed in future).
3. Now work towards solution.
4. Now, the test cases which were written in (2.) should pass and that way we can be sure that the bug has been fixed.
```

File: app/src/sharedTest/java/org/oppia/app/player/state/StateFragmentTest.kt [hunk 1]
```diff
@@ -343,6 +343,22 @@ class StateFragmentTest {
...
+    launchForExploration(TEST_EXPLORATION_ID_2).use {
+      startPlayingExploration()
+      clickContinueInteractionButton()
+      typeFractionText("1/2")
+
+      clickSubmitAnswerButton()
+
+      scrollToViewType(CONTINUE_NAVIGATION_BUTTON)
+      onView(withId(R.id.exploration_toolbar_title)).check(
+        matches(ViewMatchers.isSelected())
```
Comment (anandwana001) [inline app/src/sharedTest/java/org/oppia/app/player/state/StateFragmentTest.kt:357]: Always try to use the end method import. 
In this cases not this `import androidx.test.espresso.matcher.ViewMatchers`, it should be `import androidx.test.espresso.matcher.ViewMatchers.isSelected`

File: app/src/sharedTest/java/org/oppia/app/player/state/StateFragmentTest.kt [hunk 2]
```diff
@@ -343,6 +343,22 @@ class StateFragmentTest {
...
+  @Test
+  fun testStateFragment_loadExp_secondState_submitAnswer_clickSubmit_isToolbarTitleSelected() {
+    launchForExploration(TEST_EXPLORATION_ID_2).use {
+      startPlayingExploration()
+      clickContinueInteractionButton()
+      typeFractionText("1/2")
+
+      clickSubmitAnswerButton()
+
+      scrollToViewType(CONTINUE_NAVIGATION_BUTTON)
```
Comment (anandwana001) [inline app/src/sharedTest/java/org/oppia/app/player/state/StateFragmentTest.kt:355]: As you had described, you want to check `isSelected` after submit button click, but here you are checking after continue button click.

File: app/src/sharedTest/java/org/oppia/app/player/exploration/ExplorationActivityTest.kt [hunk 3]
```diff
@@ -629,6 +630,38 @@ class ExplorationActivityTest {
     explorationDataController.stopPlayingExploration()
   }
 
+  @Test
+  fun testMarquee_openRatioExploration_continueToInteraction_submitAnswer_checkToolBarTitleIsSelected() { // ktlint-disable max-line-length
```
Comment (anandwana001) [inline app/src/sharedTest/java/org/oppia/app/player/exploration/ExplorationActivityTest.kt:634]: it should be Toolbar not ToolBar.

**Review by anandwana001 — CHANGES_REQUESTED** [review]
We need to be more clear in which file the test will go. Writing the same test cases in both the file is not a good practice.

**Comment by prayutsu** [comment]
> As per the @rt4914 comment, You should begin with the first point. Writing test cases in the first place will not be a good idea. Also, you had only written the test cases, I didn't find the solution/fix for the issue.
> As per the tests, you had added, I am adding a review on those, which can help you afterwards when you write test cases.
> 
> ```
> 1. First try to find the exact reasons of the bugs.
> 2. Once you have identified the reason, write a test case which actually fails because of the bug. (This will be needed in future).
> 3. Now work towards solution.
> 4. Now, the test cases which were written in (2.) should pass and that way we can be sure that the bug has been fixed.
> ```

Okay, will do the same, thanks.

**Comment by prayutsu** [comment]
> We need to be more clear in which file the test will go. Writing the same test cases in both the file is not a good practice.

Okay, actually I was just trying, once I learn the correct way, I'll ensure that it is written at the right place.

File: app/src/sharedTest/java/org/oppia/app/player/state/StateFragmentTest.kt (hunk 1 above)
Comment (prayutsu) [reply app/src/sharedTest/java/org/oppia/app/player/state/StateFragmentTest.kt:357]: done

**Comment by anandwana001** [comment]
> > We need to be more clear in which file the test will go. Writing the same test cases in both the file is not a good practice.
> 
> Okay, actually I was just trying, once I learn the correct way, I'll ensure that it is written at the right place.

No issues, we can connect this more on gitter

File: app/src/sharedTest/java/org/oppia/app/player/exploration/ExplorationActivityTest.kt (hunk 3 above)
Comment (prayutsu) [reply app/src/sharedTest/java/org/oppia/app/player/exploration/ExplorationActivityTest.kt:634]: done

**Comment by prayutsu** [comment]
@anandwana001 After going through all the relevant files, I didn't find anything suspicious, so after a great struggle, I tried commenting out ```hideKeyBoard()``` in ```onSubmitButtonClicked()```. And the results are perfect as expected.
so, the culprit here is definitely 
```
inputManager.hideSoftInputFromWindow(
      fragment.view!!.windowToken,
      InputMethodManager.SHOW_FORCED
    )
```
ezgif com-video-to-gif

**Pull request closed without merging**